### ISSUE TITLE
App crashing after modifying App component in App.jsx to get tasks from mongo collection

### ISSUE THREAD

**Issue by josephgodwinkimani** [meteor/simple-todos-react#29]
I followed the Todo App with React  tutorial in [meteor.js website](https://www.meteor.com/tutorials/react/collections) and got my app crashing due to gap in line 15 to 27 which is indicated as _"...some lines skipped..."._ I tried a workaround by deleting

```
   ...some lines skipped...
       );
 }
```

It is here that now I proceed to create my to-do collection using mongo console :`db.tasks.insert({ text: "Hello world!", createdAt: new Date() });` where afterwards I got the app showing nothing (totally blank) with just styling in place.

I proceeded to adding a form, input field and a handleSubmit method to App component of App.jsx like this:

```
// imports/ui/App.jsx
import React, { Component, PropTypes } from 'react';
import ReactDOM from 'react-dom';
import { createContainer } from 'meteor/react-meteor-data';

import { Tasks } from '../api/tasks.js';
...some lines skipped...

// App component - represents the whole app
// handleSubmit method - starts here -
class App extends Component {
  handleSubmit(event) {
    event.preventDefault();

    // Find the text field via the React ref
    const text = ReactDOM.findDOMNode(this.refs.textInput).value.trim();

    Tasks.insert({
      text,
      createdAt: new Date(), // current time
    });

    // Clear form
    ReactDOM.findDOMNode(this.refs.textInput).value = '';
  }

  renderTasks() {
    return this.props.tasks.map((task) => (
      <Task key={task._id} task={task} />
```

Comparing tutorial at [meteor.js website](https://www.meteor.com/tutorials/react/collections) and the one on github which after download the zip file are different. Moreover, both Apps crashed... 

I am confused considering I have the latest meteor.js installation.

**Comment by jaredabel** [comment]
I had the same issue, not sure why the tutorial would have
...some lines skipped
As the missing 10 lines of code end up remaining rather cryptic since line numbers aren't absolute
Below is the full file as it should be at the end of Section 3
```

import React, { Component, PropTypes } from 'react';
import { createContainer } from 'meteor/react-meteor-data';

import { Tasks } from '../api/tasks.js';

import Task from './Task.jsx';

// App component - represents the whole app
class App extends Component {
  renderTasks() {
    return this.props.tasks.map((task) => (
      <Task key={task._id} task={task} />
    ));
  }

  render() {
    return (
      <div className="container">
        <header>
          <h1>Todo List</h1>
        </header>

        <ul>
          {this.renderTasks()}
        </ul>
      </div>
    );
  }
}

App.propTypes = {
  tasks: PropTypes.array.isRequired,
};

export default createContainer(() => {
  return {
    tasks: Tasks.find({}).fetch(),
  };
}, `App);
```

**Comment by andrewlintz** [comment]
I'm experiencing same issue with meteor 1.4.1 install. 
Specifically occurs between Tutorial Module: 3.1, 3.2, 3.4.

 It's >Unable to resolve some modules:  "../api/tasks.js" in  /imports/ui/App.jsx &  "../imports/api/tasks.js" in  /server/main.js

I've included the error message from OSX's terminal below.

> Unable to resolve some modules:
> 
>  "../api/tasks.js" in /Users/MyUserName/Github/2016.07.19_meteor_01/2016.09.09_meteor_React_todos/simple-todos/imports/ui/App.jsx (web.browser)
> 
> Unable to resolve some modules:
> 
>  "../imports/api/tasks.js" in /Users/MyUserName/Github/2016.07.19_meteor_01/2016.09.09_meteor_React_todos/simple-todos/server/main.js (os.osx.x86_64)
> 
> W20160910-06:54:05.088(-5)? (STDERR) /Users/MyUserName/.meteor/packages/meteor-tool/.1.4.1_1.k69xw7++os.osx.x86_64+web.browser+web.cordova/mt-os.osx.x86_64/dev_bundle/server-lib/node_modules/fibers/future.js:280
> W20160910-06:54:05.095(-5)? (STDERR)                        throw(ex);
> W20160910-06:54:05.100(-5)? (STDERR)                        ^
> W20160910-06:54:05.100(-5)? (STDERR) 
> W20160910-06:54:05.101(-5)? (STDERR) Error: Cannot find module '../imports/api/tasks.js'
> W20160910-06:54:05.101(-5)? (STDERR)     at Function.require.resolve (packages/modules-runtime/.npm/package/node_modules/install/install.js:101:1)
> W20160910-06:54:05.101(-5)? (STDERR)     at Module.resolve (packages/modules-runtime/.npm/package/node_modules/install/install.js:57:1)
> W20160910-06:54:05.101(-5)? (STDERR)     at Module.Mp.import (/Users/MyUserName/.meteor/packages/modules/.0.7.6.1aw7sk0++os+web.browser+web.cordova/npm/node_modules/reify/lib/runtime.js:61:29)
> W20160910-06:54:05.101(-5)? (STDERR)     at meteorInstall.server.main.js (server/main.js:1:8)
> W20160910-06:54:05.102(-5)? (STDERR)     at fileEvaluate (packages/modules-runtime/.npm/package/node_modules/install/install.js:153:1)
> W20160910-06:54:05.104(-5)? (STDERR)     at require (packages/modules-runtime/.npm/package/node_modules/install/install.js:82:1)
> W20160910-06:54:05.104(-5)? (STDERR)     at /Users/MyUserName/Github/2016.07.19_meteor_01/2016.09.09_meteor_React_todos/simple-todos/.meteor/local/build/programs/server/app/app.js:13:1
> W20160910-06:54:05.105(-5)? (STDERR)     at /Users/MyUserName/Github/2016.07.19_meteor_01/2016.09.09_meteor_React_todos/simple-todos/.meteor/local/build/programs/server/boot.js:292:10
> W20160910-06:54:05.105(-5)? (STDERR)     at Array.forEach (native)
> W20160910-06:54:05.114(-5)? (STDERR)     at Function._.each._.forEach (/Users/MyUserName/.meteor/packages/meteor-tool/.1.4.1_1.k69xw7++os.osx.x86_64+web.browser+web.cordova/mt-os.osx.x86_64/dev_bundle/server-lib/node_modules/underscore/underscore.js:79:11)
> => Exited with code: 1

**Comment by Phelanlewis** [comment]
I am having the same problem.

**Comment by jerrycychen** [comment]
I just pulled my old simple-todos repo from 2015 down to my new MacBook yesterday to update it to React framework. But I encountered the same issue where the todo list app on localhost:3000 wasn't showing anything with just styling in place after I entered  

`db.tasks.insert({ text: "Hello world!", createdAt: new Date() });` 

to the mongo console. So then I decided to proceed to adding a form like what @josephgodwinkimani did and then obviously my app crashed too. However the error messages showing on mine were not identical to @andrewlintz 's where the error messages were all (-8)? (STDERR) errors, so I am not entirely sure that my circumstance and solution can apply to yours. After couple hours of struggle, I realized that I forgot to enter 

`meteor npm install`

at the very beginning because I did not think of this and just jumped straight in to update it with my old codes from before starting in the middle somewhere in the tutorial. The app was fixed after I entered 

`meteor npm install`

to the terminal. Hence, I think the issues you all are experiencing might have something to do with the package dependencies. So just double check with the packages you have and have not yet installed in your repo and see if this fixes the issue. 

Hope this helps!
